internal/datacoord/services.go

```diff
@@ -1137,3 +1140,31 @@ func (s *Server) ReleaseSegmentLock(ctx context.Context, req *datapb.ReleaseSegm
 	resp.ErrorCode = commonpb.ErrorCode_Success
 	return resp, nil
 }
+
+func (s *Server) AddSegment(ctx context.Context, req *datapb.AddSegmentRequest) (*commonpb.Status, error) {
+	log.Info("DataCoord putting segment to the right DataNode",
+		zap.Int64("segment ID", req.GetSegmentId()),
+		zap.Int64("collection ID", req.GetCollectionId()),
+		zap.Int64("partition ID", req.GetPartitionId()),
+		zap.String("channel name", req.GetChannelName()),
+		zap.Int64("# of rows", req.GetRowNum()))
+	errResp := &commonpb.Status{
+		ErrorCode: commonpb.ErrorCode_UnexpectedError,
+		Reason:    "",
+	}
+	if s.isClosed() {
+		log.Warn("failed to add segment for closed server")
+		errResp.Reason = msgDataCoordIsUnhealthy(Params.DataCoordCfg.GetNodeID())
+		return errResp, nil
+	}
+	ok, nodeID := s.channelManager.getNodeIDByChannelName(req.GetChannelName())
+	if !ok {
+		log.Error("no DataNode found for channel", zap.String("channel name", req.GetChannelName()))
+		errResp.Reason = "no DataNode found for channel " + req.GetChannelName()
```

Comment:
NIT: use `fmt.Errorf` instead of string concat

Reply:
This is actually a string so I used `fmt.Sprint` instead.